src/lib.rs

```diff
@@ -307,16 +307,39 @@ pub struct HwCode {
     pub version: u16,
 }
 
+/// Determine bootstrap address based on hardware code information
+impl From<HwCode> for BootstrapAddr {
+    fn from(hwcode: HwCode) -> Self {
+        match hwcode.code {
+            0x8189 => 0x2001000,
+            _ => 0x201000,
+        }
+    }
+}
+
+type BootstrapAddr = u32;
+
 #[derive(Debug)]
 pub struct Brom {
-    address: u32,
+    pub address: BootstrapAddr,
+}
+
+impl Default for Brom {
+    fn default() -> Self {
+        // Default bootstrap address for Mediatek devices
+        Self::new(0x201000)
+    }
 }
 
 impl Brom {
-    /// Start handshake with the bootrom; The address indicates where to
-    /// load/execute the Download Agent (DA)
-    pub fn handshake(address: u32) -> impl Operation<Value = Self> {
-        HandShake::new().map(move |_| Self { address })
+    /// The address indicates where to load/execute the Download Agent (DA)
+    pub fn new(address: BootstrapAddr) -> Self {
```

Comment:
`handshake` was the constructor on purpose as that enforces from the API that you do the handshake *first* and then  any of the other operations.  So lets keep it that way :)

Reply:
Ack, fixed.